File: common/src/main/java/xyz/chunkstories/net/packets/PacketSendWorldInfo.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.net.packets;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

import xyz.chunkstories.api.net.Packet;
import xyz.chunkstories.api.net.PacketDestinator;
import xyz.chunkstories.api.net.PacketReceptionContext;
import xyz.chunkstories.api.net.PacketSender;
import xyz.chunkstories.api.net.PacketSendingContext;
import xyz.chunkstories.api.world.WorldInfo;
import xyz.chunkstories.world.WorldInfoUtilKt;

public class PacketSendWorldInfo extends Packet {
	public WorldInfo worldInfo;

	public PacketSendWorldInfo() {

	}

	public PacketSendWorldInfo(WorldInfo info) {
		this.worldInfo = info;
	}

	@Override
	public void send(PacketDestinator destinator, DataOutputStream out, PacketSendingContext ctx) throws IOException {
		out.writeUTF(WorldInfoUtilKt.serializeWorldInfo(worldInfo, false));
	}

	public void process(PacketSender sender, DataInputStream in, PacketReceptionContext processor) throws IOException {
		throw new UnsupportedOperationException();
	}
}

File: common/src/main/java/xyz/chunkstories/net/packets/PacketSendWorldInfo.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//
package xyz.chunkstories.net.packets

import xyz.chunkstories.api.Engine
import xyz.chunkstories.api.net.Packet
import xyz.chunkstories.api.player.Player
import xyz.chunkstories.api.server.UserConnection
import xyz.chunkstories.api.world.World
import xyz.chunkstories.world.serializeWorldInfo
import java.io.DataInputStream
import java.io.DataOutputStream

open class PacketSendWorldInfo(engine: Engine) : Packet(engine) {
    var worldInfo: World.Properties? = null

    constructor(engine: Engine, info: World.Properties?) : this(engine) {
        worldInfo = info
    }

    override fun send(dos: DataOutputStream) {
        dos.writeUTF(serializeWorldInfo(worldInfo!!, false))
    }

    override fun receive(dis: DataInputStream, player: UserConnection?) {
        throw UnsupportedOperationException()
    }
}